### ISSUE TITLE
Aligning SC 1.2.5 (pre-rec Audio Description) and SC 1.2.2 (pre-rec Captions) to a DOM declared language rule

### ISSUE THREAD

**Issue by wrxpirate** [w3c/wcag#4940]
I would like to propose text to address DOM declared language as a guide for how to approach accessibility assets.

If a page or component has a declared language, all accessibility assets tied to that page or component should follow that language at a minimum.

This provides clarity for content creators who may have multi-language content. 

For example:

### IF a webpage’s DOM has “en” (English) declared as it’s language, the accessibility assets must at least match that language. 
**Closed Captions:** Must be in English (if non-English audio is present, English Open Captions should also be burned into the video at non-English moments)
**Audio Description:** Must be in English (if non-English audio is present, AD must voice those English Open Captions in English to preserve content access)
                -->AD voicing a translation is the one exception for AD voicing on top of dialogue, and is intentional to preserve access.
**Transcript:** must be in English 

This does not prevent content teams from providing additional assets in the Non-English source language, just that the lang=en asset rule should be adhered to first.

**Comment by patrickhlauke** [comment]
Draft response (edited): 
<hr>
There is more nuance needed for this to be applicable to web content in the wild. 

One example: A site for learning another language which is in English but then contains videos in another language. This proposal would mandate translation (of captions/transcripts/etc) to the page's main language, which goes well beyond the intention of the current SC.

Keeping in mind again that WCAG is about accessibility - making sure that people with disabilities are not disadvantaged. If audio/video content is available in another language, PWDs are not more disadvantaged than those without disabilities.

**Comment by wrxpirate** [comment]
> Draft response: I think this lacks any necessary nuance to be applicable to the reality of web content in the wild. just as one example: what about a site for learning another language, which is in english but then contains videos in that language? what you're mandating is mandatory translation (of captions/transcripts/etc) to the page's main language, which goes well beyond the intention of the SC.
> 
> keeping in mind again that WCAG is about accessibility - making sure that people with disabilities are not disadvantaged. if audio/video content is available in another language, PWDs are not more disadvantaged than those without disabilities.

Valuable, no doubt, to think about exceptions. An English declared website, with let's say Spanish audio based video embed. In this instance (for Spanish learning), shouldn't player component and controls be declared lang=es? The "component" would have to follow the lang=es rule. Therefore, the CC and AD should be Spanish. It depends on what a user has been trained to understand for player controls and what to expect...the language would have to be declared for a screen reader to pronounce the player controls accurately. And, will it be useful at all for an early learner utilizing a screen reader? 

What I'm getting at is that the vague way that 1.2.2 and 1.2.5 adds to the confusion of what to do in multi-language scenarios. 
1.2.5 does have a bit of detail, but it is not plain language, nor easily understood: [https://www.w3.org/WAI/WCAG22/Understanding/audio-description-prerecorded.html#:~:text=The%20intent%20of%20this%20success%20criterion%20is%20to%20provide%20people%20who%20are%20blind%20or%20visually%20impaired%20access%20to%20the%20visual%20information%20in%20a%20synchronized%20media%20presentation%20in%20the%20same%20human%20language%20as%20the%20video%20or%20page%20on%20which%20it%20appears.](url)

Text of linked location above: The intent of this success criterion is to provide people who are blind or visually impaired access to the visual information in a synchronized media presentation **in the same [human language](https://www.w3.org/WAI/WCAG22/Understanding/audio-description-prerecorded.html#dfn-human-language) as the video or page on which it appears.**

Language mismatch actively blocks accessibility. Many creators are confused by what kind of assets are required on a DOM declared English facing page with a Spanish (or any other non-English language) audio video. This is 100% content loss for the blind and low vision audience if the audio description track is only Spanish Audio Description mix. This is also 100% content loss for the deaf/hard of hearing audience if there are only Spanish Closed Captions. 

Added to the mix: English declared language page, video with English _and_ Spanish audio moments. Captions should be English (and either pause for onscreen English Open Captions or continue in English translation in CC), AD and transcript should be English. The nuance is that that Spanish needs translation for the AD to read out the translation to allow for information access for the blind/low vision community. Spanish moments would also need translation for the transcript for the deafblind community.

Language access is accessibility, and if not addressed, actively blocks entire disability communities from information.

**Comment by mraccess77** [comment]
I would say caption and audio description tracks would need to indicate the language of the content with the srclang attribute.

**Comment by alastc** [comment]
@wrxpirate quoted this part of the understanding doc:
> The intent of this success criterion is to provide people who are blind or visually impaired access to the visual information in a synchronized media presentation in the same [human language](https://www.w3.org/WAI/WCAG22/Understanding/audio-description-prerecorded.html#dfn-human-language) as the video or page on which it appears.

That says video OR page, it allows for either.

Then said:
> Language mismatch actively blocks accessibility. Many creators are confused by what kind of assets are required on a DOM declared English facing page with a Spanish (or any other non-English language) audio video. This is 100% content loss for the blind and low vision audience if the audio description track is only Spanish Audio Description mix. This is also 100% content loss for the deaf/hard of hearing audience if there are only Spanish Closed Captions.

I don't follow the logic here. I agree that having video content with accessibility features not in your language is a problem, but how is it an accessibility problem? (E.g. I was on holiday in Spain, tried Netflex, and gave up because none of the dubbing / captions were in English. Even for English-origin programmes, at least the few I tried. But, I don't have a disability.)

I wasn't around when the original 1.2.x SC were created, but we've had conversations since about how language availability can be an issue. There can be some very complex scenarios, and large costs associated with doing multiple versions of  captions/AD. Overall, it was thought to be orthogonal to accessibility so we didn't go any further.

You've made some strong statements, but how does it affect with people with disabilities more than those who don't?  What's a scenario?

**Comment by wrxpirate** [comment]
> I don't follow the logic here. I agree that having video content with accessibility features not in your language is a problem, but how is it an accessibility problem? (E.g. I was on holiday in Spain, tried Netflex, and gave up because none of the dubbing / captions were in English. Even for English-origin programmes, at least the few I tried. But, I don't have a disability.)
> 
> I wasn't around when the original 1.2.x SC were created, but we've had conversations since about how language availability can be an issue. There can be some very complex scenarios, and large costs associated with doing multiple versions of captions/AD. Overall, it was thought to be orthogonal to accessibility so we didn't go any further.
> 
> You've made some strong statements, but how does it affect with people with disabilities more than those who don't? What's a scenario?

So, for your first question: the disproportionate burden of discoverability and error recovery in these scenarios provides a higher level of usability issues than for their non-disabled counterparts. I'd say this lands heavily on the side of incorrect audio description, but can be a difficult experience for the users trying to find usable closed captions as well. Audio Description users (on an English declared page) have their screen readers set to English pronunciation rules and are expecting English screen reader output. They are also expecting to find the Audio Description track in English among the player menu items and an English transcript as well if a site has it. Non-language matched Closed Captions for the deaf/hard of hearing communities also blocks this accessibility asset from use. Hearing people who can perceive that the audio is non-English, see that the captions are non-English can quickly just dismiss and understand that there are no English supports...but someone who is deaf or hard of hearing often will look for active solutions that they may already know about and spend time trying to access the audio until they can confirm that there is non-English audio. I hope that is clear....the undue time it takes for these audiences to discover that the assets are incorrect and are of no help is a complete time sink for these groups, when their non-disabled peers can just abandon the video right away because they can recognize the errors immediately.

For your example of Netflix: Being in Spain, the web page has a likely language attribute of "lang=es" (or, lang=es-ES) in which, it would be expected to match accessibility assets to Spanish. I don't think that is a conflict at all...but to be expected. Now, it certainly would be preferable for Netflix to also provide English Captions and English Audio Description, however, Netflix would be meeting the minimum bar I am recommending in this scenario...and not be required to add English assets. (when your IP is located in a place where the DOM language attribute changes to lang=en, it should flip to providing English assets for you.)

I don't want to pick on this page publisher, but this page location below is a specific example that I'm thinking of. 

The language declared as English, English copy all around the video asset, Video player controls are in English, Video asset has all French audio, English Closed Captions, and then all French AD mix. 
---> Gear icon>Audio>Track 2 for the Audio Description

[https://www.microsoft.com/en/customers/story/25382-generali-microsoft-365-copilot](url)

**For a typical hearing + seeing viewer:** no measurable impact, they just have to search for the closed captions button and enable to get English translation
**For someone who is deaf/hard of hearing:** no impact, they usually know how to find closed captions
**For someone who is blind/low vision:** Measurable time impact and difficulty confirming the error; their screen reader announces player controls in English, they finally find track 2, select it and get continued content in French with French AD. Maybe they try to fix the problem and try audio track 1 again or search elsewhere within player controls, they may look for an English transcript, or more likely: they just leave the page because they can't access the content and can't really confirm that the assets are incorrect or just missing and it has now been 5-10 minutes just trying to access.

If this same scenario had French closed captions only and French AD mix only, the impact is below:

**For a typical hearing + seeing viewer:** no measurable impact, they search for the closed captions button and see there are no other captions nor an English dub and leave or enjoy French for a while. Not ideal, and kind of a bad situation, but not much time wasted and they can confirm the error easily.
**For someone who is deaf/hard of hearing:** Measurable time impact and difficulty confirming the error; they will search for another caption track, or possibly an English transcript. They may try their own tools to ensure that there is only French audio and no translation exists. The crux is that they have no way to confirm that the audio is incorrect or the captions are incorrect.
**For someone who is blind/low vision:** Measurable time impact and difficulty confirming the error; their screen reader announces player controls in English, they finally find track 2, select it and get continued content in French with French AD. Maybe they try to fix the problem and try audio track 1 again or search elsewhere within player controls, they may look for an English transcript, or more likely: they just leave the page because they can't access the content and can't really confirm that the assets are incorrect or just missing and it has now been 5-10 minutes just trying to access.


What I am proposing:

Page DOM declares lang=en
(video with only French audio, English player controls, English copy surrounding)
All accessibility assets should follow that declaration at a minimum. (English)
CC == English translation (Ideally, these should also be burned in as Open Captions)
AD == English AD mix (this is tricky, but AD should be acting as an English dub in this scenario as per AD standards)
Transcript == English translation
---->site can choose to also add French closed captions, a French AD mix, and a French transcript, but not required.

Sorry that is long....sort of one of these complicated things that should have been a presentation :)

**Comment by alastc** [comment]
> For your example of Netflix: Being in Spain, the web page has a likely language attribute of "lang=es" (or, lang=es-ES) 

I don't think so, I was logged in and had an English interface. I didn't check the markup, but the text of the interface was in English, which I expected as it knows where I'm from (or detected the language setting in my browser).

> in which, it would be expected to match accessibility assets to Spanish. 

So already not the case, Netflix would have to provide assets for all possible languages people could login with, even if they don't have permission to do so from the rights-holders.

@patrickhlauke - I think the draft response above is still applicable.

**Comment by wrxpirate** [comment]
I agree that WCAG does not require translating all assets for all possible languages. That would clearly fall outside the scope of accessibility requirements.

What I am trying to highlight is a slightly different problem: situations where an accessibility feature technically exists but is not usable by the audience implied by the page language, which creates a disproportionate burden for users relying on accessibility mechanisms.

WCAG already acknowledges the importance of language context for assistive technologies.

For example:

WCAG 3.1.1 – Language of Page requires the primary language of a page to be programmatically determinable so assistive technologies can correctly interpret and present content.

WCAG 3.1.2 – Language of Parts requires language changes within content to be programmatically identified so assistive technologies can switch pronunciation rules and processing appropriately.

These criteria exist because language context directly affects how assistive technologies behave.

In the scenario I described:

The page declares lang="en"
The interface and player controls are English
A user navigating with a screen reader is operating in an English language context

In that environment, a user who selects an accessibility asset such as captions or Audio Description will reasonably expect that asset to function within the same language context unless it is clearly labeled otherwise.

When the accessibility feature itself is in a different language than the page context, the experience differs depending on disability.

A sighted hearing user encountering a video with non-English audio can immediately detect the mismatch and decide whether to continue watching or leave.

A user relying on accessibility features may need to:

Locate the accessibility control
Enable the feature
Discover the language mismatch only after activation
Attempt additional troubleshooting before concluding that the asset is unusable

In that situation the accessibility mechanism becomes a dead end, which increases discovery and error-recovery time for users who depend on that feature.

My suggestion about aligning accessibility assets with the declared language of the page is intended as a practical way to reduce these failure scenarios for assistive technology users, rather than as a requirement to translate all media content to all languages.

I agree that broader questions about localization and content distribution fall outside WCAG, but the usability of accessibility mechanisms themselves seems directly relevant to accessibility outcomes.
